generators/client-2/templates/src/main/webapp/app/admin/user-management/_user-management.state.ts

```diff
@@ -1,152 +1,154 @@
-                        };
-                    }
-                }
-            }).result.then(function() {
-                $state.go('user-management', null, { reload: true });
-            }, function() {
-                $state.go('user-management');
-            });
-        }]
-    })
-    .state('user-management.edit', {
-        url: '/{login}/edit',
-        data: {
-            authorities: ['ROLE_ADMIN']
-        },
-        onEnter: ['$stateParams', '$state', '$uibModal', function($stateParams, $state, $uibModal) {
-            $uibModal.open({
-                templateUrl: 'app/admin/user-management/user-management-dialog.html',
-                controller: 'UserManagementDialogController',
-                controllerAs: 'vm',
-                backdrop: 'static',
-                size: 'lg',
-                resolve: {
-                    entity: ['User', function(User) {
-                        return User.get({login : $stateParams.login});
-                    }]
-                }
-            }).result.then(function() {
-                $state.go('user-management', null, { reload: true });
-            }, function() {
-                $state.go('^');
-            });
-        }]
-    })
-    .state('user-management.delete', {
-        url: '/{login}/delete',
-        data: {
-            authorities: ['ROLE_ADMIN']
-        },
-        onEnter: ['$stateParams', '$state', '$uibModal', function($stateParams, $state, $uibModal) {
-            $uibModal.open({
-                templateUrl: 'app/admin/user-management/user-management-delete-dialog.html',
-                controller: 'UserManagementDeleteController',
-                controllerAs: 'vm',
-                size: 'md',
-                resolve: {
-                    entity: ['User', function(User) {
-                        return User.get({login : $stateParams.login});
-                    }]
-                }
-            }).result.then(function() {
-                $state.go('user-management', null, { reload: true });
-            }, function() {
-                $state.go('^');
-            });
+    <%_ } _%>
+
+};
+export const userMgmtDetailState = {
+    name: 'user-management-detail',
+    parent: 'admin',
+    url: '/user/:login',
+    data: {
+        authorities: ['ROLE_ADMIN'],
+        pageTitle: 'user-management.detail.title'
+    },
+    views: {
+        'content@': { component: UserMgmtDetailComponent }
+    },
+    resolve: {
+        translatePartialLoader: ['$translate', '$translatePartialLoader', function ($translate, $translatePartialLoader) {
+            $translatePartialLoader.addPart('user-management');
+            return $translate.refresh();
         }]
-    });
-}
+    }
+};
+export const userMgmtNewState = {
+    name: 'user-management.new',
+    url: '/new',
+    data: {
+        authorities: ['ROLE_ADMIN']
+    },
+    // onEnter: ['$stateParams', '$state', '$uibModal', function($stateParams, $state, $uibModal) {
```

Comment:
you need to do as below, what do you mean the component is not specified? can you check if below works?
```
    onEnter: ['$transition$', (trans: Transition) => {
        private $state: StateService = trans.injector().get('StateService');
        private modalService: NgbModal = trans.injector().get('NgbModal');
        modalService.open(UserManagementDialogComponent).result.then(result => {
           $state.go('user-management', null, { reload: true });
        }, (reason) => {
           $state.go('user-management');
        });
    }],
```